plugins/item_previews/web_client/views/ItemPreviewWidget.js

```diff
@@ -1,103 +1,179 @@
-import ItemCollection from 'girder/collections/ItemCollection';
+import _ from 'underscore';
+import $ from 'jquery';
 import View from 'girder/views/View';
-import { restRequest } from 'girder/rest';
+import { restRequest, apiRoot } from 'girder/rest';
+import { _whenAll } from 'girder/misc';
 
 import ItemPreviewWidgetTemplate from '../templates/itemPreviewWidget.pug';
+import ItemPreviewItemTemplate from '../templates/itemPreviewItem.pug';
 import '../stylesheets/itemPreviewWidget.styl';
 
 /**
  * The Item Preview widget shows a preview of items under a given folder.
- * For now, the only supported item previews are image previews.
+ * For now, this only supports json and images file types.
  */
 var ItemPreviewWidget = View.extend({
     events: {
         'click a.g-item-preview-link': function (event) {
             var id = this.$(event.currentTarget).data('id');
             this.parentView.itemListView.trigger('g:itemClicked', this.collection.get(id));
-        },
-        'click .g-widget-item-previews-wrap': function (event) {
-            this.wrapPreviews = !this.wrapPreviews;
-            this.render();
-            if (this.wrapPreviews) {
-                this.$('.g-widget-item-previews-container').addClass('g-wrap');
-            } else {
-                this.$('.g-widget-item-previews-container').removeClass('g-wrap');
-            }
         }
     },
 
     initialize: function (settings) {
-        this.collection = new ItemCollection();
+        this._isSupportedItem = this._isSupportedItem.bind(this);
```

Comment:
This self-modifying behavior is too tricky, IMO. Most Backbone methods are smart enough to call things with the correct `this` bound. If there are truly problematic places, `_.bind` should be used inline.